### PR TITLE
Rename ActiveFedoraPidBasedJob to ActiveFedoraIdBasedJob

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

File: sufia-models/app/jobs/active_fedora_id_based_job.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class ActiveFedoraIdBasedJob
+  def queue_name
+    :pid_based
+  end
+
+  attr_accessor :pid
```
Comment (mjgiarlo) [inline sufia-models/app/jobs/active_fedora_id_based_job.rb]: WHAAAAAAAAAAAAAAAAT?  ;)

**Comment by mjgiarlo** [comment]
:+1:

File: sufia-models/app/jobs/active_fedora_id_based_job.rb (hunk 1 above)
Comment (jcoyne) [inline sufia-models/app/jobs/active_fedora_id_based_job.rb]: That seems like a follow up change. I'll do that too if this is merged.

File: sufia-models/app/jobs/active_fedora_pid_based_job.rb [hunk 2]
```diff
@@ -1,22 +1,7 @@
...
-
-  alias_method :generic_file, :object
-  alias_method :generic_file_id, :pid
-
-  def run
-    raise RuntimeError, "Define #run in a subclass"
+class ActiveFedoraPidBasedJob < ActiveFedoraIdBasedJob
+  extend Deprecation
+  def self.extended(document)
+    Deprecation.warn ActiveFedoraPidBasedJob, "ActiveFedoraPidBasedJob is deprecated; use ActiveFedoraIdBasedJob instead."
```
Comment (awead) [inline sufia-models/app/jobs/active_fedora_pid_based_job.rb]: If you make this change to the 5.x branch, you could go ahead and remove this in 6.0.

**Pull request merged**